nilearn/glm/tests/test_regression.py

```diff
@@ -1,32 +1,38 @@
 """Test functions for models.regression"""
 
-import numpy as np
+import pytest
 from numpy.testing import assert_almost_equal, assert_array_almost_equal
 
 from nilearn.glm import ARModel, OLSModel
 
-RNG = np.random.RandomState(42)
-X = RNG.standard_normal(size=(40, 10))
-Y = RNG.standard_normal(size=(40,))
 
+@pytest.fixture()
+def X(rng):
+    return rng.standard_normal(size=(40, 10))
 
-def test_OLS():
+
+@pytest.fixture()
+def Y(rng):
+    return rng.standard_normal(size=(40, 10))
+
+
+def test_OLS(X, Y):
     model = OLSModel(design=X)
     results = model.fit(Y)
     assert results.df_residuals == 30
     assert results.residuals.shape[0] == 40
     assert results.predicted.shape[0] == 40
 
 
-def test_AR():
+def test_AR(X, Y):
     model = ARModel(design=X, rho=0.4)
     results = model.fit(Y)
     assert results.df_residuals == 30
     assert results.residuals.shape[0] == 40
     assert results.predicted.shape[0] == 40
 
 
-def test_residuals():
+def test_residuals(X, Y):
     Xintercept = X.copy()
```

Comment:
I think the copy was there because the tests we using globals, with fixtures this should not be needed, no?